### DISCUSSION TITLE
RADOS permission denied (error connecting to the cluster)

### DISCUSSION THREAD

**Discussion by osaffer** [rook/rook#8831] [Q&A] [answered]
Dear,

Last week we had a datastore corruption. Despite the VM have been restored, we have not been able to restore a good state of the cluster.

All OSD are up : 

ceph osd stat  
15 osds: 15 up (since 2d), 15 in (since 2d); epoch: e20787

What is strange too, it that ceph operator is not able to recreate the MDS

How may I fix this kind of problem ?

health: HEALTH_ERR
            1 filesystem is degraded
            1 filesystem is offline
            1 mds daemon damaged

From crash collector , two nodes show the following error message : 

RADOS permission denied (error connecting to the cluster)

Thanks a lot of for your help.

Regards

**Comment by leseb** [comment, marked as answer]
This is a dup of #8745, which was fixed in Rook v1.7.4. Please upgrade your Rook version.

**Comment by osaffer** [comment]


**Reply by leseb** [reply]
Upgrading the Rook operator, does not always restart the Ceph daemons. Just upgrading the Rook operator is fine. It will fix the MDS issue.

**Reply by osaffer** [reply]
Hi,

I performed an upgrade to 1.7.4

I applied common and crds. For crds I was obliged to delete for the crds the line with spec.preserveUnknownFields to make the apply successful.

Now I have : 
5622142976 LastUpdated:2021-09-27T14:22:32Z} Versions:0xc001f57600}
2021-09-27 14:22:41.449301 I | op-osd: OSD 0 is not ok-to-stop. will try updating it again later
2021-09-27 14:22:42.560240 I | op-osd: OSD 1 is not ok-to-stop. will try updating it again later
2021-09-27 14:22:43.729206 I | op-osd: OSD 10 is not ok-to-stop. will try updating it again later
2021-09-27 14:22:44.828675 I | op-osd: OSD 11 is not ok-to-stop. will try updating it again later
2021-09-27 14:22:45.945767 I | op-osd: OSD 15 is not ok-to-stop. will try updating it again later
2021-09-27 14:22:47.052667 I | op-osd: OSD 16 is not ok-to-stop. will try updating it again later
2021-09-27 14:22:48.164307 I | op-osd: OSD 17 is not ok-to-stop. will try updating it again later
2021-09-27 14:22:49.231401 I | op-osd: OSD 2 is not ok-to-stop. will try updating it again later
2021-09-27 14:22:50.317121 I | op-osd: OSD 3 is not ok-to-stop. will try updating it again later


ceph health detail
HEALTH_ERR 1 filesystem is degraded; 1 filesystem is offline; 1 mds daemon damaged; mons k,n,p are low on available space; 1 pool(s) have no replicas configured
[WRN] FS_DEGRADED: 1 filesystem is degraded
    fs postgresfs is degraded
[ERR] MDS_ALL_DOWN: 1 filesystem is offline
    fs postgresfs is offline because no MDS is active for it.
[ERR] MDS_DAMAGE: 1 mds daemon damaged
    fs postgresfs mds.0 is damaged
[WRN] MON_DISK_LOW: mons k,n,p are low on available space
    mon.k has 20% avail
    mon.n has 15% avail
    mon.p has 15% avail
[WRN] POOL_NO_REDUNDANCY: 1 pool(s) have no replicas configured
    pool 'replicapool' has no replicas configured

For information all OSD are up, All pgs and objects are healthy at 100%
Thanks again for your help

**Reply by osaffer** [reply]
rook-ceph-crashcollector-adakub1.do.gov.lu      req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-crashcollector-adakub2.do.gov.lu      req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-crashcollector-adakub3.do.gov.lu      req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-crashcollector-adakub4.do.gov.lu      req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-crashcollector-adakub5.do.gov.lu      req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-mds-postgresfs-a      req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-mds-postgresfs-b      req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-mgr-a         req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-mon-k         req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-mon-n         req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-mon-p         req/upd/avl: 1/1/1      rook-version=v1.7.4
rook-ceph-osd-0         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-1         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-10        req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-11        req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-15        req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-16        req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-17        req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-2         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-3         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-4         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-5         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-6         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-7         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-8         req/upd/avl: 1/1/1      rook-version=v1.5.0
rook-ceph-osd-9         req/upd/avl: 1/1/1      rook-version=v1.5.0
